### PR TITLE
MCR-1562 DOIs with registrant codes

### PR DESCRIPTION
**Pull request by sebhofmann**

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 360642abf0c2830f35c7acca6cf4efbbc801a48b! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/1103-pull_request-196

File: mycore-pi/src/test/java/org/mycore/pi/doi/MCRDOIParserTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        Optional<MCRPersistentIdentifier> parsedDOIOptional = parser.parse(doi);
+
+        Assert.assertTrue("DOI should be parsable!", parsedDOIOptional.isPresent());
+
+        MCRDigitalObjectIdentifier parsedDOI = (MCRDigitalObjectIdentifier) parsedDOIOptional.get();
+
+        Assert.assertEquals("DOI Prefix should match!", expectedPrefix,
+            parsedDOI.getPrefix());
+
+        Assert.assertEquals("DOU Suffix should match", expectedSuffix,
```
Comment (yagee-de) [inline mycore-pi/src/test/java/org/mycore/pi/doi/MCRDOIParserTest.java]: DOU -> DOI

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 642fb69c0f412b3346f0a0352894e90b578a5bb3! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/1111-pull_request-196

**Review by yagee-de — APPROVED** [review]

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 642fb69c0f412b3346f0a0352894e90b578a5bb3! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/1111-pull_request-196

**Pull request merged**